### ISSUE TITLE
change app icon need delay above one minute and app background in Android

### ISSUE THREAD

**Issue by charles0122** [chandrabezzo/flutter_dynamic_icon_plus#4]
![image](https://github.com/chandrabezzo/flutter_dynamic_icon_plus/assets/70085736/cc1d2a3d-a912-4669-8986-91a2b8f67f4d)
after one minute 

 
![image](https://github.com/chandrabezzo/flutter_dynamic_icon_plus/assets/70085736/16edc3ee-4e47-419f-8321-58f2af0ae3ca)

flutter sdk: 3.19.6 
dev OS: windows11
If you don't let the app run in the background, the modification won't be triggered

**Comment by charles0122** [comment]
![image](https://github.com/chandrabezzo/flutter_dynamic_icon_plus/assets/70085736/3adb84d8-8471-46b4-93a8-9d1c68eaa85d)
![image](https://github.com/chandrabezzo/flutter_dynamic_icon_plus/assets/70085736/862da584-3d50-48ff-a13e-083921e6bdc4)

**Comment by charles0122** [comment]
I also tried executing with the example of main, the same result

**Comment by chandrabezzo** [comment]
Yes @charles0122 this plugin will be triggered if you close the app. I implement this approach to minimize bad user experience to our application because the app will force closed and need to reopen again by user manually.

**Comment by shixiwen000** [comment]
> Yes @charles0122 this plugin will be triggered if you close the app. I implement this approach to minimize bad user experience to our application because the app will force closed and need to reopen again by user manually.

It would be nice if it was controllable.

**Comment by shixiweneee** [comment]
> Yes @charles0122 this plugin will be triggered if you close the app. I implement this approach to minimize bad user experience to our application because the app will force closed and need to reopen again by user manually.

It will be nice if it is controllable.

**Comment by chandrabezzo** [comment]
Hi @shixiwen000 can you describe about controllable? Like triggerring any button we can change the app icon?

**Comment by LouisRaverdy** [comment]
Hello, I have the same issue, on android the icon only change when the app is fully closed (not on background). If I understand well, this is wanted, but what you you mean by "minimize bad user experience" ? And is this possible to have the choice for changing the icon immediately ?

Thanks !

**Comment by charles0122** [comment]
@LouisRaverdy I think "minimize bad user experience" means that after the user switches the icon, they don't have to force the app to restart, but rather make the change naturally, such as the next time they open the app. Maybe you can prompt the user that the icon change will take effect the next time they start the app. If forced changes are required, I think there should be a configuration item

**Comment by charles0122** [comment]
@LouisRaverdy  Even if there is no project that provides configuration for forced restart, you can also write your own prompt to the user to change the operation and restart the app, and then write your own restart app logic

**Comment by LouisRaverdy** [comment]
Okay thanks I'll implement that ! Actually, there are 2 big issues with the Android version, I fixed them in my fork but It's wired that nobody noticed them.
